File: typescript-playwright/src/helpers/AuthHelper.ts
import { type APIRequestContext } from "@playwright/test";
import { AuthApi } from "../api/AuthApi.ts";
import { ADMIN_USERNAME, ADMIN_PASSWORD } from "../utils/constants.ts";

export class AuthHelper {
  private readonly authApi: AuthApi;

  constructor(request: APIRequestContext) {
    this.authApi = new AuthApi(request);
  }

  async getAdminToken(): Promise<string> {
    return this.authApi.getToken(ADMIN_USERNAME, ADMIN_PASSWORD);
  }

  async getAdminHeaders(): Promise<Record<string, string>> {
    return this.authApi.getAuthHeaders(ADMIN_USERNAME, ADMIN_PASSWORD);
  }

  async loginWithInvalidPassword(
    times: number,
    password = "wrong_password",
  ): Promise<void> {
    for (let i = 0; i < times; i++) {
      await this.authApi.login({
        username: ADMIN_USERNAME,
        password: password,
      });
    }
  }
}

File: python/src/helpers/auth_helper.py
from __future__ import annotations

import requests

from src.api.auth_api import AuthApi
from src.utils.constants import ADMIN_PASSWORD, ADMIN_USERNAME


class AuthHelper:
    def __init__(self, session: requests.Session) -> None:
        self._auth_api = AuthApi(session)

    def get_admin_token(self) -> str:
        return self._auth_api.get_token(ADMIN_USERNAME, ADMIN_PASSWORD)

    def get_admin_headers(self) -> dict[str, str]:
        return self._auth_api.get_auth_headers(ADMIN_USERNAME, ADMIN_PASSWORD)
